modules/datahandling/filerequests.go

```diff
@@ -313,8 +312,10 @@ func (f fileChangeRequest) process(db dbfs.DBFS) ([]dhClosure, error) {
 		return []dhClosure{toSenderClosure{msg: messages.NewEmptyResponse(messages.StatusUnauthorized, f.Tag)}}, nil
 	}
 
+	prevChanges, err := db.PullChanges(fileMeta)
```

Comment:
this prev changes should really be happening inside of `db.CBAppendFileChanges`.